frontend/src/app/main/data/plugins.cljs

```diff
@@ -8,13 +8,23 @@
   (:require
    [app.common.data.macros :as dm]
    [app.main.data.modal :as modal]
+   [app.main.data.notifications :as ntf]
    [app.main.store :as st]
    [app.plugins.register :as preg]
    [app.util.globals :as ug]
    [app.util.http :as http]
+   [app.util.i18n :as i18n :refer [tr]]
+   [app.util.time :as dt]
    [beicon.v2.core :as rx]
    [potok.v2.core :as ptk]))
 
+(defn save-plugin-permissions-peek
+  [id permissions]
+  (ptk/reify ::save-plugin-permissions-peek
+    ptk/UpdateEvent
+    (update [_ state]
+      (assoc-in state [:profile :props :plugins-permissions-peek :data id] permissions))))
```

Comment:
I think is not a good idea to use profile props to store this, firstly because this property is easy to be overwritten on single fetch of profile; i mean is a backend synchronized property.
Also you don't need to creeate so nested data structure. I mean the all global state is profile oriented, so you don't need at all neest it under profile, you can create a new, specific to the plugins permisions key on the root of the state, making it more simple to access. Also, looks like `:data` only contains the `:id`, can this be stored directly under parent key without neesting it under `:data`?

Reply:
I put this there because is where the plugins info is stored. This is kind of a plugins permissions cache, so it made sense to have it next to the real plugins info